src/main/kotlin/ar/edu/unahur/obj2/socios/Barrio.kt

```diff
@@ -0,0 +1,27 @@
+package ar.edu.unahur.obj2.socios
+
+interface Barrio{
+
+    fun aplicarPropinaPorBarrio(propina: Double) : Double
+}
+
+object LasRosas : Barrio {
+
+    override fun aplicarPropinaPorBarrio(propina: Double) = propina.plus(50.0)
+}
+
+object LasLauchas : Barrio{
+
+    override fun aplicarPropinaPorBarrio(propina: Double) = propina.div(2)
+}
+
+object LasTorres : Barrio {
+
+    override fun aplicarPropinaPorBarrio(propina: Double) = propina
+}
+
+object BarrioVerde: Barrio{
+
+    override fun aplicarPropinaPorBarrio(propina: Double) = if(propina <= 200.0) 200.0 else propina
```

Comment:
Acá hubieran usado `max()` que devuelve el valor más alto. Si fuera menor al limite (200), devuelve el limite. Si fuera mayor devuelve el valor de la propina.
```suggestion
    override fun aplicarPropinaPorBarrio(propina: Double) = max(propina, 200.0)
```